### PR TITLE
Use Checkstyle last version

### LINKED ISSUE #3355: [Bug] Cannot update com.puppycrawl.tools:checkstyle due to conflict
### Description

We cannot bump com.puppycrawl.tools:checkstyle from 10.12.1 to a future version because of a conflict with `com.google.collections:google-collections`:

### Reproduction

1. Update `com.puppycrawl.tools:checkstyle` in `build.gradle` to >10.12.1
1. `./gradlew checkstyleMain`

See this error message

```
Execution failed for task ':checkstyleMain'.
> Could not resolve all files for configuration ':checkstyle'.
   > Could not resolve com.google.guava:guava:32.0.1-jre.
     Required by:
         project : > com.puppycrawl.tools:checkstyle:10.12.3
      > Module 'com.google.guava:guava' has been rejected:
           Cannot select module with conflict on capability 'com.google.collections:google-collections:32.1.2-jre' also provided by [com.google.collections:google-collections:1.0(runtime)]
   > Could not resolve com.google.collections:google-collections:1.0.
     Required by:
         project : > com.puppycrawl.tools:checkstyle:10.12.3 > org.apache.maven.doxia:doxia-core:1.12.0 > org.codehaus.plexus:plexus-container-default:2.1.0
      > Module 'com.google.collections:google-collections' has been rejected:
           Cannot select module with conflict on capability 'com.google.collections:google-collections:1.0' also provided by [com.google.guava:guava:32.1.2-jre(jreRuntimeElements)]
```

### Exit Criteria

- [x] Fix the conflict and merge an update to the latest version of `com.puppycrawl.tools:checkstyle`

### Additional Context

- https://github.com/opensearch-project/security/pull/3345
- https://github.com/opensearch-project/security/pull/3105

### PR DESCRIPTION
**Pull request by davidosorno**
### Description
Update Checkstyle to the latest version to resolve conflicts .

### Issues Resolved
- Resolves #3355 

### Check List
- [ ] New functionality includes testing
- [ ] New functionality has been documented
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -32,6 +32,7 @@ buildscript {
         jjwt_version = '0.11.5'
         guava_version = '32.1.2-jre'
         jaxb_version = '2.3.8'
+        checkstyle_version = '10.12.4'
```
Comment (peternied) [inline build.gradle:35]: Nit: Why create a variable, its only used once?

---

Whenever I say `Nit` this is a take it or leave it comment.  You don't need to make a change for these types of suggestions if you don't want to and you can resolve this comment thread without a response if you so desire.  Also you can write a detailed response and make changes.

Nit comments are personal preference based and while they might be useful to consider/adopt they shouldn't block productivity.

**Review by peternied — APPROVED** [review]
Nicely done!

**Comment by peternied** [comment]
@davidosorno Could you please rebase this change off of the latest from main?  It will fix the CI issues that are failing

File: build.gradle [hunk 2]
```diff
@@ -639,7 +640,7 @@ dependencies {
     integrationTestImplementation 'com.unboundid:unboundid-ldapsdk:4.0.14'
 
     //Checkstyle
-    checkstyle 'com.puppycrawl.tools:checkstyle:10.12.1'
+    implementation "com.puppycrawl.tools:checkstyle:${checkstyle_version}"
```
Comment (cwperks) [inline build.gradle:643]: I see that we also have an entry in the gradle plugins section that adds the [checkstyle gradle plugin](https://github.com/opensearch-project/security/blob/main/build.gradle#L65).

Could this entry in the dependencies section be redundant?

File: build.gradle (hunk 1 above)
Comment (davidosorno) [reply build.gradle:35]: Good point. 
I adopted this practice long time ago, so I will reconsider it. Thanks a lot @peternied for the feedback.

File: build.gradle (hunk 2 above)
Comment (davidosorno) [reply build.gradle:643]: @cwperks it seems likely, I removed the line 643 and it worked.
should I remove it and rebase the PR or is there any other way I can test it?

**Pull request closed without merging**